src/scmdata/units.py

```diff
@@ -129,3 +134,10 @@ def target(self) -> str:
         Target unit
         """
         return self._target
+
+
+def get_unit_registry() -> openscm_units.ScmUnitRegistry:
```

Comment:
Why use getter rather than a global?

Reply:
The getter makes testing easier. You can patch the getter, but with a module attribute you need to do something like:
```
orig_registry = scmdata.units.UNIT_REGISTRY

try:
    scmdata.units.UNIT_REGISTRY = new_registry
    # Do something
finally:
    scmdata.units.UNIT_REGISTRY = orig_registry
```

Reply:
It is also easier to swap out behaviour in future. `get_unit_registry` can do something at runtime such as read from a file or use an env variable.

Reply:
> It is also easier to swap out behaviour in future. get_unit_registry can do something at runtime such as read from a file or use an env variable.

True

> You can patch the getter, but with a module attribute you need to do something like

I think you can just do (https://stackoverflow.com/questions/38336090/mocking-a-global-variable)

```
from mock import patch
@patch('scmdata.units.UNIT_REGISTRY', new_registry) 
```

Reply:
Nice. I didn't know that. I thought you could only patch callables.

Having a quick play around, I believe the downside of that is that you can't use `assert_called` etc so might work in some places, but not all

Reply:
You can use it on the methods (I think it returns a MagicMock so you can do assert_called etc. on all the methods)